iroha/test_network/src/lib.rs

```diff
@@ -642,7 +642,11 @@ impl TestClientConfiguration for ClientConfiguration {
     fn test(api_url: &str) -> Self {
         let mut configuration = ClientConfiguration::from_path(CLIENT_CONFIGURATION_PATH)
             .expect("Failed to load configuration.");
-        configuration.torii_api_url = api_url.to_owned();
+        if !api_url.starts_with("http") {
+            configuration.torii_api_url = "http://".to_owned() + api_url;
+        } else {
+            configuration.torii_api_url = api_url.to_owned();
+        }
```

Comment:
This looks like a temporary fix - its is ok for local testing but not for our dev branch to my mind.

Reply:
It is a test library. I think its code quality doesnt matter that much

Reply:
We need to be able to maintain it - same as any library or tests. Otherwise we will be searching for these problems and spending a lot of time on fixing it later. Any library should be transparent.

Reply:
It will not take long to correct.

Reply:
To my mind there are two possible ways there:
1. Either you assume and add http to everything (and give back error if there is some other schema already)
2. Or you allow schema specification

It is possible to have a default http:
But then it is common in rust libs to do 2 methods - 1 with default, 1 without

This string matching and replacing is rather error prone and just seems to me like a temporary fix.

We already specify schema in our configs - let's have the same behavior in tests. The less tests replace - the better. We should only replace there some necessary things for isolating cases.

Reply:
Excuse me,
I think "http" hard-coding does not matter, except for temporary cases like now.
Whether add certificate and encode to https or not is more "outer" issue, isn't it?

Reply:
> Excuse me, I think "http" hard-coding does not matter, except for temporary cases like now. Whether add certificate and encode to https or not is more "outer" issue, isn't it?

`http` hardcoding is not a problem to my mind. I am more concerned about us having a different style of API url specification in configs and in tests. And also that tests add this under the hood. It seems to me as with all the configs we should just replace it all now with proper urls with http specified.

Reply:
My opinion is that if we follow something like TDD, tests should specify behavior, not approximate it. Divirgence from main logic behavior is general is unwanted.

Reply:
Ah, this code does smth like an implicit type conversion
`(Host or Url) -> Url`
and that's the point, right?

Reply:
Yep